normandy/base/utils.py

```diff
@@ -50,7 +50,14 @@ def get_client_ip(request):
 
 
 def filter_m2m(qs, field, values):
-    qs.annotate(_count=Count(field)).filter(_count=len(values))
+    """
+    Filters a queryset by an exact list of many to many relations.
+    """
+    values = list(values)
+
+    qs = qs.annotate(_count=Count(field)).filter(_count=len(values))
+
+    if len(values):
+        qs = qs.filter(**{'{}__in'.format(field): values})
```

Comment:
it's the combination of line 58 (which checks the count) and this line 61 which make sure it's an exact match.